server/src/main/java/com/objectcomputing/checkins/services/memberprofile/MemberProfileServices.java

```diff
@@ -6,7 +6,7 @@
 public interface MemberProfileServices {
     MemberProfile getById(UUID id);
 
-    Set<MemberProfile> findByValues(String name, String role, UUID pdlId, String workEmail);
+    Set<MemberProfile> findByValues(String name, String role, String pdlId, String workEmail);
```

Comment:
@meyerinj why was pdlId changed to string from UUID ? I ask because I started converting checkin notes and got hit with an error : 
```
Internal Server Error: SQL Error executing Query: ERROR: operator does not exist: character varying = uuid
  Hint: No operator matches the given name and argument type(s). You might need to add explicit type casts.
  Position: 67
```
I am assuming this error is because in our Postgres SQL script we have defined UUID as varchar because I think there some issues defining it as UUID, can't remember. @mkimberlin is that correct ?

I found a solution and it works, adding a cast to datatype resolves that error.

```

CREATE CAST (varchar AS uuid)
  WITH INOUT
  AS IMPLICIT;
```

My question is, do we create a new sql script to create the table again and add this cast or there is a way to alter the table and add the cast, can't find that on how to alter? @meyerinj @mkimberlin

Reply:
Postgres doesn't play nice with UUIDs that is why all the ids are stored as varchar. Then in the entity classes we have this line on the id field    @TypeDef(type = DataType.STRING) .  This has been working.

Reply:
@hwilliams314 it doesn't work when you convert findbyfields() method to use @Query annotation approach. You get an error that I posted.

Reply:
@bhatiam12 To clarify, you converted the UUID param on findByCheckinid and findByCreatedbyid to a String and it caused an exception?

Reply:
@meyerinj I didn't convert it to string, it is UUID. here is the code : 
```
  @Query(" SELECT * " +
            "FROM checkin_notes cn " +
            "WHERE (:checkinid IS NULL OR cn.checkinid = :checkinid) " +
            "AND (:createbyid IS NULL OR cn.createdById = :createbyid) ")
    Set<CheckinNote> search(UUID checkinid, UUID createbyid);
```

Reply:
Okay, I think I understand. Your suggestion is that rather than changing the repository to use a String we should leave it as a UUID and use the cast code that posted above?

Reply:
yup, correct. was this error the reason that you changed repository to use string or was it something else ?

Reply:
It was. I'm not opposed to using this workaround instead of changing the data type in the Java code.

Reply:
I think another option would be to cast in the query: `OR cn.checkinid::text =`

Reply:
let me try casting in the query.